services/address_book_service.proto

```diff
@@ -29,37 +27,132 @@ import "transaction_response.proto";
- * 3. When signed and submitted, the server will verify that account 2 keys have signed, and the keys on the operator account have signed.
- * Hedera council should have ability to make all edits in addition to add/remove nodes
+ * The Address Book service provides the ability for Hedera network node
+ * administrators to add, update, and remove consensus nodes. This addition,
+ * update, or removal of a consensus node requires governing council approval,
+ * but each node operator may update their own operational attributes without
+ * additional approval, reducing overhead for routine operations.
+ *
+ * Most operations are `privileged operations` and require governing council
+ * approval.
+ *
+ * ### For a node creation transaction.
+ * - The node operator SHALL create a `createNode` transaction.
+ *    - The node operator SHALL sign this transaction with the active `key` for
+ *      the account to be assigned as the "node account".
+ *    - The node operator MUST deliver the signed transaction to the Hedera
+ *      council representative.
+ *    - The Hedera council representative SHALL arrange for council members to
+ *      review and sign the transaction.
+ *    - Once sufficient council members have signed the transaction, the
+ *      Hedera council representative SHALL submit the transaction to the
+ *      network.
+ * - Upon receipt of a valid and signed node creation transaction the network
+ *   software SHALL
+ *    - Validate the threshold signature for the Hedera governing council
+ *    - Validate the signature of the active `key` for the account to be
+ *      assigned as the "node account".
+ *    - Create the new node in a pending state.
+ *    - When executing the next `freeze` transaction with `freeze_type` set to
+ *      `PREPARE_UPGRADE`, update network configuration and transition the
+ *      new pending node to an active status.
+ *
+ * ### For a node deletion transaction.
+ * - The node operator or Hedera council representative SHALL create a
+ *   `deleteNode` transaction.
+ *    - If the node operator creates the transaction
+ *       - The node operator MUST sign this transaction with the active `key`
+ *         for the account assigned as the "node account".
+ *       - The node operator SHALL deliver the signed transaction to the Hedera
+ *         council representative.
+ *    - The Hedera council representative SHALL arrange for council members to
+ *      review and sign the transaction.
+ *    - Once sufficient council members have signed the transaction, the
+ *      Hedera council representative SHALL submit the transaction to the
+ *      network.
+ * - Upon receipt of a valid and signed node deletion transaction the network
+ *   software SHALL
+ *    - Validate the threshold signature for the Hedera governing council
+ *    - Validate the signature of the active `key` for the account assigned
+ *      as the "node account", _if present_.
```

Comment:
Validate the signature on the transaction?
Not sure what isn't possible here...

Reply:
I am not sure if validating a key _if present_ is possible. We can define the account key as optional, but this will only tell us whether there is a valid signature or not. We cannot distinguish between an invalid signature and a missing signature. I believe this distinction would be required here.

Reply:
If the signature is present, we can certainly validate that, and if not present not validate the signature.  I think I see your point; the wording is a bit confusing, then again the process is a bit confusing too (because it might be initiated in two different ways).
I'll rework the wording a bit to make the necessary actions a bit clearer.